ocaml/xapi/xapi_pool_update.ml

```diff
@@ -120,47 +169,52 @@ let with_api_errors f x =
   baseurl=url
   ${signed:+gpgkey=file:///etc/pki/rpm-gpg/key}
 *)
-let create_yum_config ~__context ~self ~url ~location =
+let create_yum_config ~__context ~self ~url =
   let key = Db.Pool_update.get_key ~__context ~self in
   let signed = String.length key <> 0 in
   let signed_index = if signed then 1 else 0 in
   let name_label = Db.Pool_update.get_name_label ~__context ~self in
-  let yum_config = location ^ "/yum.conf" in
-  let oc = open_out yum_config in
-  Printf.fprintf oc "[main]\nkeepcache=0\nreposdir=/dev/null\ngpgcheck=%d\nrepo_gpgcheck=%d\n\n" signed_index signed_index;
-  Printf.fprintf oc "[%s]\nname=%s\nbaseurl=%s\n" name_label name_label url;
-  if signed then Printf.fprintf oc "gpgkey=file:///etc/pki/rpm-gpg/%s" key;
-  close_out oc
-
-let attach ~__context ~self ~host =
+  (Printf.sprintf ("[main]\nkeepcache=0\nreposdir=/dev/null\ngpgcheck=%d\nrepo_gpgcheck=%d\n\n") signed_index signed_index)
+  ^(Printf.sprintf ("[%s]\nname=%s\nbaseurl=%s\n") name_label name_label url)
+  ^(if signed then Printf.sprintf ("gpgkey=file:///etc/pki/rpm-gpg/%s") key else "")
+
+let attach_helper ~__context ~uuid ~vdi =
+  Mutex.execute updates_to_attach_count_tbl_mutex 
+  ( fun () ->
+    let mount_point_parent_dir = String.concat "/" [Xapi_globs.host_update_dir; uuid] in
+    let mount_point = Filename.concat mount_point_parent_dir "vdi" in
+    let host = Helpers.get_localhost ~__context in
+    let ip = try Db.Host.get_address ~__context ~self:host with _ -> "127.0.0.1" in
+    let url = "http://" ^ ip ^ Constants.get_pool_update_download_uri ^ uuid ^ "/vdi" in
+    let count = try Hashtbl.find updates_to_attach_count_tbl uuid with _ -> 0 in
+    if count = 0 then begin
+      debug "pool_update.attach_helper %s to %s" uuid mount_point;
+      if (try Sys.is_directory mount_point with _ -> false) then detach_helper ~__context ~uuid ~vdi;
+      let output, _ = Forkhelpers.execute_command_get_output "/bin/mkdir" ["-p"; mount_point] in
+      debug "pool_update.attach_helper Mountpoint created (output=%s)" output;
+      if vdi_visible ~__context ~vdi ~host then begin
```

Comment:
I think this is unnecessary. If the VDI isn't visible, message_forwarding has gone wrong.